### DISCUSSION TITLE
Adding special labels for the NXF_VER env variable

### DISCUSSION THREAD

**Discussion by abhi18av** [nextflow-io/nextflow#1892] [Ideas]
Hi team,

I was wondering whether we could simplify the process of helping users to test their code against the latest `stable` and `edge` version.

The proposal is to add two special labels `stable` and `edge` which would automatically resolve to the latest stable and edge release from Github.

For example,
```
NXF_VER=edge nextflow run nextflow-io/hello
```

I think that this could be an additive change, which retains the current ability to specify the exact version which the user would like to use.

**Comment by abhi18av** [comment]
I guess, this might be an alternative for the suggested feature.

```
export NXF_EDGE=1
curl get.nextflow.io | bash

```

**Comment by drpatelh** [comment]
I think something like this would be really useful given that pipelines may have to rely on edge releases more. Would be much easier to document too and would bypass the need to find and export the exact release version manually.

**Reply by pditommaso** [reply]
Having NXF_EDGE=1 you can use the usual `nextflow self-update` command to update it.

**Reply by abhi18av** [reply]
One possible use-case which comes to mind is if the user would like to have both the stable as well as the edge version? 

As of now, I think, there are two ways to use the edge release

1. Download from Github releases and save it along side the `stable` release
2. Use the `NXF_EDGE=1` release, this would overwrite the current NF binary

**Reply by drpatelh** [reply]
Does `NXF_EDGE` exist already? For me, `NXF_VER=edge` would be more intuitive or maybe `NXF_VER_EDGE` because it's more explicit about the Nextflow version.

**Reply by pditommaso** [reply]
umm maybe already works like that .. don't remember 😆 

https://github.com/nextflow-io/nextflow/blob/8f274c9551d3dd3c6ed34c3e0b7fe01bf2b5decc/nextflow#L226
